src/cdomains/apron/apronDomain.apron.ml

```diff
@@ -728,6 +775,27 @@ struct
     let j_env = Environment.lce x_env y_env in
     A.change_environment Man.mgr join_c j_env false
 
+  (* TODO: Deduplicate code shared with join *)
+  let join_interval x y =
+    let x = to_interval x in
+    let y = to_interval y in
+    let r = if DLiftInterval.is_bot x then
+      y
+    else if DLiftInterval.is_bot y then
+      x
+    else (
+      let x_env = A.env x in
+      let y_env = A.env y in
+      let c_env = gce x_env y_env in
+      let x_c = A.change_environment IntervalManager.mgr x c_env false in
+      let y_c = A.change_environment IntervalManager.mgr y c_env false in
+      let join_c = A.join IntervalManager.mgr x_c y_c in
+      let j_env = Environment.lce x_env y_env in
+      let r = A.change_environment IntervalManager.mgr join_c j_env false in
+      r
+    ) in
+    of_interval r
```

Comment:
Why does it repeat all the `join_interval` logic from `DLift` instead of just delegating to that `join_interval` in the middle (like `join` does on common environment)?

Reply:
Changed it, you may look at it again.